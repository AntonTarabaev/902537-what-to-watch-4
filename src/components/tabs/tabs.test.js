import Tabs from "@components/tabs/tabs";

const PageTabs = [
  {
    id: `Overview`,
    component: <div>0</div>,
  },
  {
    id: `Details`,
    component: <div>1</div>,
  },
  {
    id: `Reviews`,
    component: <div>3</div>,
  },
];

describe(`FilmDetailsTabs component render correctly`, () => {
  it(`on first tab`, () => {
    const tree = renderer.create(
        <Tabs
          tabs={PageTabs}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`on second tab`, () => {
    const component = renderer.create(
        <Tabs
          tabs={PageTabs}
        />
    );

    component.root.findAllByProps({className: `movie-nav__link`})[1].props.onClick({preventDefault: () => {}});

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`on third tab`, () => {
    const component = renderer.create(
        <Tabs
          tabs={PageTabs}
        />
    );

    component.root.findAllByProps({className: `movie-nav__link`})[2].props.onClick({preventDefault: () => {}});

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
