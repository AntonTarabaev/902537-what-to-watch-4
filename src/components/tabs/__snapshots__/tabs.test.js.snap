// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmDetailsTabs component render correctly on first tab 1`] = `
Array [
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href={null}
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>,
  <div>
    0
  </div>,
]
`;

exports[`FilmDetailsTabs component render correctly on second tab 1`] = `
Array [
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href={null}
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>,
  <div>
    1
  </div>,
]
`;

exports[`FilmDetailsTabs component render correctly on third tab 1`] = `
Array [
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href={null}
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>,
  <div>
    3
  </div>,
]
`;
